parser/conceptParser.go

```diff
@@ -305,24 +326,48 @@ func collectAllPArseErrors(results []*ParseResult) (errs []ParseError) {
 	return
 }
 
-func validateConcepts(conceptDictionary *gauge.ConceptDictionary) *ParseResult {
+func ValidateConcepts(conceptDictionary *gauge.ConceptDictionary) *ParseResult {
+	res := &ParseResult{ParseErrors: []ParseError{}}
+	var conceptsWithError []*gauge.Concept
 	for _, concept := range conceptDictionary.ConceptsMap {
 		err := checkCircularReferencing(conceptDictionary, concept.ConceptStep, nil)
 		if err != nil {
-			return &ParseResult{CriticalErrors: []ParseError{err.(ParseError)}, FileName: concept.FileName}
+			delete(conceptDictionary.ConceptsMap, concept.ConceptStep.Value)
```

Comment:
Deleting from conceptDictionary. However, when gauge is started in daemon mode, two caches for concept are maintained, conceptDictionary and conceptsCache. Deleting from one cache should update the other too.